tests/testthat/test-clean-names.R

```diff
@@ -1,42 +1,197 @@
-  expect_equal(names(clean)[17], "leadingpunct") # for testing alternating cases below with e.g., case = "upper_lower"
-  expect_equal(names(clean)[18], "average_number_of_days") # for testing alternating cases below with e.g., case = "upper_lower"
-  expect_equal(names(clean)[19], "jan2009sales") # no separator around number-word boundary if not existing already
-  expect_equal(names(clean)[20], "jan_2009_sales") # yes separator around number-word boundary if it existed
+test_that("All scenarios for make_clean_names", {
+  expect_equal(
+    make_clean_names("sp ace"),
+    "sp_ace"
+  )
+  expect_equal(
+    make_clean_names(c("repeated", "repeated", "REPEATED")),
+    paste0("repeated", c("", "_2", "_3"))
+  )
+  expect_equal(
+    make_clean_names("a**^@"),
+    "a"
+  )
+  expect_equal(
+    make_clean_names("%"),
+    "percent"
+  )
+  expect_equal(
+    make_clean_names("*"),
+    "x"
+  )
+  expect_equal(
+    make_clean_names("!"),
+    "x"
+  )
+  expect_equal(
+    make_clean_names(c("*", "!")),
+    c("x", "x_2")
+  )
+  expect_equal(
+    make_clean_names("d(!)9"),
+    "d_9"
+  )
+  expect_equal(
+    make_clean_names("can\"'t"),
+    "cant"
+  )
+  expect_equal(
+    make_clean_names("hi_`there`"),
+    "hi_there"
+  )
+  expect_equal(
+    make_clean_names("  leading spaces"),
+    "leading_spaces"
+  )
+  expect_equal(
+    make_clean_names("€"),
+    "x"
+  )
+  expect_equal(
+    make_clean_names("ação", ascii=FALSE),
+    "acao"
+  )
+  expect_equal(
+    make_clean_names("ação"),
+    "acao"
+  )
+  expect_equal(
+    make_clean_names("Farœ"),
+    "faroe"
+  )
+  expect_equal(
+    make_clean_names("a b c d e f"),
+    "a_b_c_d_e_f"
+  )
+  expect_equal(
+    make_clean_names("testCamelCase"),
+    "test_camel_case"
+  )
+  expect_equal(
+    make_clean_names("!leadingpunct"),
+    "leadingpunct"
+  )
+  expect_equal(
+    make_clean_names("average # of days"),
+    "average_number_of_days"
+  )
+  expect_equal(
+    make_clean_names("jan2009sales"),
+    "jan2009sales"
+  )
+  expect_equal(
+    make_clean_names("jan 2009 sales"),
+    "jan_2009_sales"
+  )
+  expect_equal(
+    make_clean_names("not_first_unicode_µ"),
+    "not_first_unicode_m"
+  )
+  expect_equal(
+    make_clean_names("µ_first_unicode"),
+    "m_first_unicode"
+  )
+  expect_equal(
+    make_clean_names("a/b"),
+    "a_b",
+    info="No custom replacement"
+  )
+  expect_equal(
+    make_clean_names("a/b", replace=c("/"="_per_")),
+    "a_per_b",
+    info="Custom replacement"
+  )
 })
 
+testing_vector <-
+  c(
+    "sp ace",
+    "repeated",
+    "a**^@",
+    "%",
+    "*",
+    "!",
+    "d(!)9",
+    "REPEATED",
+    "can\"'t",
+    "hi_`there`",
+    "  leading spaces",
+    "€",
+    "ação",
+    "Farœ",
+    "a b c d e f",
+    "testCamelCase",
+    "!leadingpunct",
+    "average # of days",
+    "jan2009sales",
+    "jan 2009 sales",
+    "not_first_unicode_µ",
+    "µ_first_unicode"
+  )
+result_vector_noascii <-
```

Comment:
Are these result vectors used again?  I don't see them

Reply:
I had intended to use them, but I ended up going down a different path.  I've removed them.